python/ray/data/tests/test_object_gc.py

```diff
@@ -235,6 +237,28 @@ def consume(p):
     assert "Spilled" not in meminfo, meminfo
 
 
+def test_global_bytes_spilled(shutdown_only):
+    # The object store is about 90MB.
+    ctx = ray.init(object_store_memory=90e6)
+    # The size of dataset is 500*(80*80*4)*8B, about 100MB.
+    ds = ray.data.range_tensor(500, shape=(80, 80, 4), parallelism=100).materialize()
+
+    check_no_spill(ctx, ds.repeat(), spill_expected=True)
```

Comment:
nit: alternatively, can we leave `check_no_spill` unmodified, and instead check that `check_no_spill` fails the assertion?
```
with pytest.raises(AssertionError):
    check_no_spill(ctx, ds.repeat())
```